### PR TITLE
Fix #106: Handle complex floating-point types.

### LINKED ISSUE #106: Handle complex floating-point types
Built-in C support for complex numbers, e.g. `float _Complex` or `double _Complex`, is currently not handled by dstep.

Support looks like it should be reasonably straightforward to implement.  Complex numbers are defined by an enum value in `clang.c.Index`:
https://github.com/jacob-carlborg/dstep/blob/561545d48cdc9ab7fe71efc359df456c138e126e/clang/c/Index.d#L2838

and so adding support would probably need to involve adding another special-case to the `switch` statement in `dstep.Type.translateType`:
https://github.com/jacob-carlborg/dstep/blob/561545d48cdc9ab7fe71efc359df456c138e126e/dstep/translator/Type.d#L55

Note, a special-case `translateComplex` would be needed because it will be necessary to extract the base floating-point type of the complex number(s) concerned.

### PR DESCRIPTION
**Pull request by ciechowoj**
As in the title.

### CODE REVIEW

**Comment by jacob-carlborg** [comment]
Are there any literals for Complex in C?

**Comment by ciechowoj** [comment]
It doesn't seem so, however there are constants for imaginary unit `_Imaginary_I`, `_Complex_I`,` I`.

**Comment by jacob-carlborg** [comment]
I googled this and found code like `double complex z1 = 1.0 + 3.0 * I` [1]. Will that work in D?

[1] http://stackoverflow.com/questions/6418807/how-to-work-with-complex-numbers-in-c

**Comment by ciechowoj** [comment]
Not without changes, however dstep doesn't translate complex (complicated) expressions at the moment anyway. E.g.

```
assertTranslates(
q"C
static const int x = 1 * 2;

C",
q"D
extern (C):

extern __gshared const int x;
D");
```

No expression.

**Comment by jacob-carlborg** [comment]
Right, that's not needed.

**Comment by jacob-carlborg** [comment]
Could you please update the changelog as well.

File: dstep/translator/Type.d [hunk 1]
```diff
@@ -236,6 +240,16 @@ body
         return translateType(context, type.kind, rewriteIdToObjcObject);
 }
 
+string translateComplex (Type type)
+{
+    switch (type.element.kind) {
```
Comment (jacob-carlborg) [inline dstep/translator/Type.d]: Curly braces on the next line.

File: changelog.md [hunk 2]
```diff
@@ -40,6 +40,7 @@
 * [Issue 59](https://github.com/jacob-carlborg/dstep/issues/59): Shouldn't dstep exit with status code when there is some kind of error
 * [Issue 83](https://github.com/jacob-carlborg/dstep/issues/83): New multiline translation
 * [Issue 85](https://github.com/jacob-carlborg/dstep/issues/85): dstep not converting `const T x[]` to `const (T)* x`
+* [Issue 106](https://github.com/jacob-carlborg/dstep/issues/106): Handle complex floating-point types.
```
Comment (jacob-carlborg) [inline changelog.md]: This is a new feature, it should be under `New/Changed Features`.

Comment (ciechowoj) [inline changelog.md]: Done.

**Comment by jacob-carlborg** [comment]
👌

**Comment by jacob-carlborg** [comment]
Thanks.

**Pull request merged**